ledger-api/grpc-definitions/com/daml/ledger/api/v1/transfer.proto

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "TransferOuterClass";
+option java_package = "com.daml.ledger.api.v1";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1";
+
+// Complete view of an on-ledger transfer.
+message Transfer {
+
+  // Assigned by the server. Useful for correlating logs.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string update_id = 1;
+
+  // TODO this field (whether it is populated) can signal whether it is originates from the same participant or not (
+  //  so additional retry etc procedures are in order or not), on the other hand with multi-hosted parties this is not
+  //  so obvious.
+  // The ID of the command which resulted in this transfer. Missing for everyone except the submitting party.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string command_id = 2;
+
+  // The workflow ID used in command submission. Only set if the ``workflow_id`` for the command was set.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 3;
+
+  // The absolute offset. The format of this field is described in ``ledger_offset.proto``.
+  // Required
+  string offset = 4;
+
+  // Party on whose behalf the command was executed.
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string submitter = 5;
+
+  oneof event {
+    TransferredOutEvent transfer_out_event = 7;
+    TransferredInEvent transfer_in_event = 8;
+  }
+}
+
+// Records that a contract has been transferred-out, and it becomes unusable on the source domain
+message TransferredOutEvent {
+
+  // The ID of the transfer-out event. This needs to be used as an input for a transfer-in TransferCommand.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string transfer_out_id = 1;
```

Comment:
What is the difference between the `transfer_out_id` and the `update_id`?

Reply:
Oh yes, I was going to ask you that :) If this is the same as the update_id of the transfer, then we could drop this from here, and rename for the TransferInCommand for like `transfer_out_update_id`. WDYT?

Reply:
In the current Canton implementation, we can make the two the same. But there are also reasons for keeping them separate. For comparison with UTXO ledgers:
* The update ID identifies a UTXO transaction
* The transfer-out-id is a  transaction output in UTXO, like a contract ID is a transaction output of the transaction. 
Transactions themselves cannot be spent in UTXO, but transaction outputs can. In general, a UTXO transaction can have many transaction outputs (and this is also the case in Daml with created contracts). For transfer-out transactions, we now have the special case that the transfer out will have only one spendable output.

So while it is technically feasible to unify update id and transfer-out id, conceptually it makes sense to keep them separate.